### PR TITLE
Jonmv/concrete document field validation

### PR DESCRIPTION
**Pull request by jonmv**
@baldersheim please review.

### CODE REVIEW

**Review by baldersheim — DISMISSED** [review]
Looks good, only nitpick Arrays.asList => List.of.

File: document/src/test/java/com/yahoo/document/datatypes/ArrayTestCase.java [hunk 1]
```diff
@@ -217,10 +217,10 @@ public void testWrappedList() {
         assertEquals(new StringFieldValue("apple"), subArray.get(1));
 
 
-        assertEquals(false, array.containsAll(Arrays.<StringFieldValue>asList(new StringFieldValue("bob"))));
-        assertEquals(true, array.containsAll(Arrays.<StringFieldValue>asList(new StringFieldValue("foo"), new StringFieldValue("boo"), new StringFieldValue("apple"))));
+        assertEquals(false, array.containsAll(Arrays.asList(new StringFieldValue("bob"))));
+        assertEquals(true, array.containsAll(Arrays.asList(new StringFieldValue("foo"), new StringFieldValue("boo"), new StringFieldValue("apple"))));
```
Comment (jonmv) [inline document/src/test/java/com/yahoo/document/datatypes/ArrayTestCase.java:221]:
```suggestion
        assertEquals(false, array.containsAll(List.of(new StringFieldValue("bob"))));
        assertEquals(true, array.containsAll(List.of(new StringFieldValue("foo"), new StringFieldValue("boo"), new StringFieldValue("apple"))));
```

File: document/src/test/java/com/yahoo/document/datatypes/ArrayTestCase.java [hunk 2]
```diff
@@ -217,10 +217,10 @@ public void testWrappedList() {
...
 
-        array.removeAll(Arrays.<StringFieldValue>asList(new StringFieldValue("foo"), new StringFieldValue("boo")));
+        array.removeAll(Arrays.asList(new StringFieldValue("foo"), new StringFieldValue("boo")));
```
Comment (jonmv) [inline document/src/test/java/com/yahoo/document/datatypes/ArrayTestCase.java:223]:
```suggestion
        array.removeAll(List.of(new StringFieldValue("foo"), new StringFieldValue("boo")));
```

File: document/src/test/java/com/yahoo/document/datatypes/ArrayTestCase.java [hunk 3]
```diff
@@ -249,7 +249,7 @@ public void testWrappedList() {
             assertFalse(it.hasNext());
         }
 
-        array.addAll(Arrays.<StringFieldValue>asList(new StringFieldValue("microsoft"), new StringFieldValue("google")));
+        array.addAll(Arrays.asList(new StringFieldValue("microsoft"), new StringFieldValue("google")));
```
Comment (jonmv) [inline document/src/test/java/com/yahoo/document/datatypes/ArrayTestCase.java:252]:
```suggestion
        array.addAll(List.of(new StringFieldValue("microsoft"), new StringFieldValue("google")));
```

**Review by baldersheim — APPROVED** [review]

**Pull request merged**